src/modvault/moditem.py

```diff
@@ -0,0 +1,211 @@
+import os
+import util
+from PyQt5 import QtCore, QtWidgets, QtGui
+from downloadManager import IconCallback
+import urllib.parse
+
+
+class ModView(QtCore.QObject):
+    """
+    Helps with displaying mods in the mod widget. Handles updates to view
+    unrelated to underlying data, like downloading mod previews. Forwards
+    interaction with the view.
+    """
+    mod_double_clicked = QtCore.pyqtSignal(object)
+
+    def __init__(self, model, view, delegate, dler):
+        QtCore.QObject.__init__(self)
+        self._model = model
+        self._view = view
+        self._delegate = delegate
+        self._dler = dler
+
+        self._view.setModel(self._model)
+        self._view.setItemDelegate(self._delegate)
+        self._delegate.mod_preview_missing.connect(self.download_mod_preview)
+        self._view.doubleClicked.connect(self._mod_double_clicked)
+        self._view.viewport().installEventFilter(self._delegate.tooltip_filter)
+
+    def download_mod_preview(self, mod):
+        name = os.path.basename(urllib.parse.unquote(mod.thumbnail))
+        cb = IconCallback(name, self._mod_preview_downloaded)
+        self._dler.downloadModPreview(mod.thumbnail, name, cb)  # (url, name, requester)
+
+    # TODO make it a utility function?
+    def _model_items(self):
+        model = self._model
+        for i in range(model.rowCount(QtCore.QModelIndex())):
+            yield model.index(i, 0)
+
+    def _mod_preview_downloaded(self, modname, icon):
+        for idx in self._model_items():
+            mod = idx.data().mod
+            if mod.thumbnail != "":
+                name = os.path.basename(urllib.parse.unquote(mod.thumbnail))
+                if name.lower() == modname.lower():  # Previews are not case-preserving
+                    self._view.update(idx)
+
+    def _mod_double_clicked(self, idx):
+        self.mod_double_clicked.emit(idx.data().mod)
+
+
+class ModItemDelegate(QtWidgets.QStyledItemDelegate):
```

Comment:
That's a bit of duplication of games widget code, but I guess we'll figure out a way to factor out common parts in the future.

Reply:
well, I started with copying the game widget code and transformed it into mod widget code.
(That's the way I learn how things work.)